sdks/python/apache_beam/runners/direct/executor.py

```diff
@@ -434,9 +440,10 @@ class VisibleExecutorUpdate(object):
     raise an exception.
     """
 
-    def __init__(self, exception=None):
-      self.finished = exception is not None
-      self.exception = exception
+    def __init__(self, exc_info=(None, None, None)):
+      self.finished = exc_info[0] is not None
+      self.exception = exc_info[1] or exc_info[0]
```

Comment:
This translates both (ExnInstance, None, None) and (ExnType, ExnInstance, Traceback) into ExnInstance.